src/Sulu/Component/Security/Authorization/AccessControl/AccessControlManager.php

```diff
@@ -163,7 +163,7 @@ private function getUserObjectPermission(SecurityCondition $securityCondition, U
     private function getUserObjectPermissionByArray($permissions, UserInterface $user)
     {
         if (empty($permissions)) {
-            return [];
+            return null;
```

Comment:
Maybe this change is not needed, if we make sure that a return value of an empty array always means that no permissions are set. This means that if we make sure that [the ContentRepository](https://github.com/sulu/sulu/blob/e58a64c/src/Sulu/Component/Content/Repository/ContentRepository.php#L854) returns an array with all permission values set false instead of returning an empty array, then this could work. This would have to look similar as in the [SecuritySubscriber](https://github.com/sulu/sulu/blob/e58a64c35406e1d0b02fda64a34c81071af11d31/src/Sulu/Component/Content/Document/Subscriber/SecuritySubscriber.php#L96-L105).

https://github.com/sulu/sulu/blob/e58a64c35406e1d0b02fda64a34c81071af11d31/src/Sulu/Component/Content/Document/Subscriber/SecuritySubscriber.php#L96-L105